external_parser/parse_example_binary.cc

```diff
@@ -148,8 +148,13 @@ bool binary_parser::parse_examples(vw *all, v_array<example *> &examples) {
 
     auto reward_function_info =
         flatbuffers::GetRoot<v2::RewardFunctionInfo>(payload);
```

Comment:
if `_default_reward` is a member of `example_joiner` instead of `example_binary_parser` then it could be set here i.e. `example_joiner.set_default_reward(reward_function_info->default_reward())` and then it could just be used in `process_joined` instead of being passed in